lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,275 @@
+    #
+    # Proxy responses:
+    #   Response if IP is already reserved:
+    #     {"code": 200, "success": true, "data": [{"id":"157","subnetId":"32","ip":"10.10.10.1","is_gateway":null,"description":null,"hostname":null,"mac":null,"owner":null,"tag":"2","deviceId":null,"location":null,"port":null,"note":null,"lastSeen":null,"excludePing":"0","PTRignore":"0","PTR":"0","firewallAddressObject":null,"editDate":null,"customer_id":null}],"time":0.008}
+    #   Response if IP address is available
+    #     {"code": 200, "success": false, "message": "No addresses found", "time": 0.007}
+    #   Response if missing required parameters:
+    #     {"code": 400, "error": ["A 'cidr' parameter for the subnet must be provided(e.g. 100.10.10.0/24)", "Missing 'ip' parameter. An IPv4 address must be provided(e.g. 100.10.10.22)"]}
+    #   Response if subnet not exists:
+    #     {"code": 200, "success": false, "message": "No subnets found", "time": 0.007}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def ip_exists(ip, subnet, externalipam_group)
+      raise "subnet cannot be nil" if subnet.nil?
+      response = parse get("/subnet/#{subnet}/#{ip}?group=#{URI.escape(externalipam_group.to_s)}")
+      raise ::Foreman::Exception.new(N_(response['error'])) if response['error']
+      (response['code'] == 200 && response['message'] && response['message'] == 'No addresses found') ? false : true
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain IP address for subnet_id %s from External IPAM."), subnet)
+    end
+
+    # Frees up an IP address from a given subnet in External IPAM.
+    #
+    # Inputs: 1. ip:     IP address to be freed up
+    #         2. subnet: The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24", IPv6 - "2001:db8:abcd:12::/124")
+    #
+    # Returns: true if IP is deleted successfully from External IPAM, otherwise false.
+    #
+    # Proxy responses:
+    #   Response if success:
+    #     {"code": 200, "success": true, "message": "Address deleted", "time": 0.012}
+    #   Response if subnet error:
+    #     {"code": 200, "error": "The specified subnet does not exist in External IPAM."}
+    #   Response if IP already deleted:
+    #     {"code": 200, "success": false, "message": "No addresses found", "time": 0.006}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def delete_ip_from_subnet(ip, subnet, externalipam_group)
+      raise "subnet cannot be nil" if subnet.nil?
+      response = parse delete("/subnet/#{subnet}/#{ip}?group=#{URI.escape(externalipam_group.to_s)}")
+      raise ::Foreman::Exception.new(N_(response['error'])) if response['error']
+      (response['code'] == 200 && response['message'] && response['message'] == 'Address deleted') ? true : false
```

Comment:
Please always check only HTTP response code, check and fix in all REST methods. Other implementations will likely just return 200, they'd need to find hard what exact message to set.